app/controllers/conference_controller.rb

```diff
@@ -21,10 +25,40 @@ def schedule
     conf_end = 20
     @conf_period = conf_end - @conf_start
 
-    if @dates == Date.current
-      @today = Date.current.strftime('%Y-%m-%d')
+    @day = Time.find_zone(@conference.timezone).today
+    # the schedule takes you to today if it is a date of the schedule
+    if @dates.include? @day
+      @has_date = true
+      current_hour = DateTime.now.in_time_zone(@conference.timezone).strftime('%H').to_i
+      # the schedule takes you to today hour or to next day start hour if the
+      # conference end time has been reached
+      if current_hour <= conf_end
+        @hour_column = current_hour > @conf_start ? current_hour - @conf_start : 0
+      else
+        @hour_column = 0
+        @day += 1.day
+      end
     else
-      @today = @conference.start_date.strftime('%Y-%m-%d')
+      @day = @dates.first
+    end
+  end
+
+  def allevents
+    @dates = @conference.start_date..@conference.end_date
+
+    @scheduled_events = @conference.program.events.scheduled.order(:start_time)
+    @unscheduled_events = @conference.program.events.unscheduled
+
+    day = Time.find_zone(@conference.timezone).today
```

Comment:
The difference is that `today` has the time and `day` doesn't have it.

I think I should change the names of the variables to make it more clear (`day_with_time` instead of `today`) and also `today` is equivalent to that:

`today = Time.find_zone(@conference.timezone).now`

which is similar to how `day` is get so I think I should change that too.